### ISSUE TITLE
Project

### ISSUE THREAD

**Issue by ChandlerGotfried** [ChandlerGotfried/LGBT_Proj#1]
@AaronGullickson I am struggling to connect my current project to GitHub, I am not sure what is going on right now

**Comment by ChandlerGotfried** [comment]
@AaronGullickson never mind, I figured it out. I had to redownload the file and change my password. I wanted to know if you could help me re-code and combine my data so I could turn 5 variables into a single variable?

**Comment by AaronGullickson** [comment]
@ChandlerGotfried Sorry Chandler. Just seeing this. Do you still need help with this?

**Comment by ChandlerGotfried** [comment]
@AaronGullickson I don't anymore, I realized that I do not want that data and I am just going to use the sexual identity variable. But there is one thing I am struggling with. I made boolean statements for two variables so I could us them in graphs but for some reason I still have NAs in the data. How can I remove them? I tried on my own but it turned all my data into NAs. Also, if I have two variables with multiple NAs, do I have to take out all NAs from both variables in order to compare them? If so what is the easiest way?

**Comment by AaronGullickson** [comment]
If you take out all NAs you are doing casewise deletion, which is simple but can lead to bias and will shrink your sample size. The best solution would be multiple imputation with `mice`. However, if the number of missing values is small, it may not be worth it. See the slides on the section on missing data. 

To remove all cases with missing values on some `var1` and `var2`:

```r
mydata <- subset(mydata, !is.na(var1) & !is.na(var2))
```

**Comment by ChandlerGotfried** [comment]
@AaronGullickson Also my data has been inputted into RStudio. It is in organize_data.R as both my raw data and my subsetted data. If it is not in the right spot I will change it.

**Comment by ChandlerGotfried** [comment]
@AaronGullickson is there anyway you could help me figure out how to make Boxplots or other graphs? I seem to be getting a lot of errors.

**Comment by AaronGullickson** [comment]
@ChandlerGotfried Happy to but you need to point me to your code so I can see whats not working. Right now I am not seeing any boxplots anywhere in the code you have up on GitHub.

**Comment by ChandlerGotfried** [comment]
@AaronGullickson I added a boxplot in analysis > organize_data so you can take a look. I am just not understanding if I have to recode some of my variables. They are all in numbers with the NAs inputted for other numbered variables. I just don't know if I should leave them numbered or actually give them word variables (i.e "1" recoded to "none").

**Comment by ChandlerGotfried** [comment]
@AaronGullickson Also I uploaded a barplot to the organize_data file underneath the boxplot and I am unable to get the gay marriage plot to work. It is at 80%/20% but it should be more like 95%/5%.
Clarification: its the comparative barplot

**Comment by AaronGullickson** [comment]
@ChandlerGotfried On the boxplot, q15a is recorded as a categorical likert scale variable and the "y" variable in the comparative boxplot needs to be quantitative. You would need to turn it into a quantitative score variable from 1-6. You can do that pretty easily with `as.numeric`, so: 

```r
boxplot(as.numeric(q15a)~q14a, data=sjsd, na.rm=TRUE,
        names=c("None","Some","About half","Most","All"), 
        xlab="Coming out to family", ylab="Level of upport from family")
```

However, there are supposed to be 8 and 9 values in their for missing values. It appears that those have already been coded as NAs because they don't appear in the data, but always be sure to check stuff like this. 

ideally, you should recode q15a into a new variable with a more intuitive name that checks for missing values and maybe converts to numeric values if you like. Also, when you recode q14a, recode it into a new variable with a more intuitive name. This will allow you to check yourself before you wreck yourself but crosstabbing the new variable with the old variable to make sure everything is in the right place.

**Comment by AaronGullickson** [comment]
@ChandlerGotfried The barplot looks like 95/5 to me. Its just that the x-axis doesn't go out far enough. You can set the `xlim` argument to make sure the axis goes to where you want to, like so:

```r
barplot(percent, main="Support of LGBT in the military", horiz=TRUE, names.arg = c("No", "Yes"),
         xlab="Percent", xlim=c(0,100))
```

Or better yet, you can use ggplot and not have to worry about these details.

**Comment by ChandlerGotfried** [comment]
@AaronGullickson would it be possible for you to help me figure out how to source my data into the product file > paper file > tablefigs.Rmd? I figure it is pretty easy but want to make sure it is well done

**Comment by AaronGullickson** [comment]
@ChandlerGotfried 

I would not bother with the tablefigs.Rmd file. Just put your tables and figures directly into main.Rmd. The syntax for the pathway is to use ".." to move back a folder, so it would be:

```r
load("../../analysis/output/analytical_data.RData")
```

**Comment by ChandlerGotfried** [comment]
@AaronGullickson Ok so far I have some graphs and other smaller things. I am not able to knit my project and I also wanted to know if you could help me set up a model for my project. What I want to do is on there but I am just not sure if I am on the right track.

**Comment by ChandlerGotfried** [comment]
@AaronGullickson when I use varwidth in ggplot boxplots, when the box is wider does that mean that there is more data for that variable? Say attending an event was the data variable, if the line on the box plot is wider for never attended than attended once a month, more people responded to never attended?

**Comment by AaronGullickson** [comment]
@ChandlerGotfried Yes, wider boxplots mean larger numbers in that category.

**Comment by ChandlerGotfried** [comment]
@AaronGullickson Ok so far I have some graphs and other smaller things. I am not able to knit my project and I also wanted to know if you could help me set up a model for my project. What I want to do is on there but I am just not sure if I am on the right track.

**Comment by AaronGullickson** [comment]
@ChandlerGotfried A couple of problems which I have fixed in commit 18269765d8cdbc26d9086245ec0f810acbe6d4ae. 

1. You were copying the same setup code chunk but not changing the name "setup" so you had duplicate names in your code chunks. 
2. You were using `include=FALSE` in all of your code chunks so they don't show up.
3. You had removed the figured height and width from the top-level header causing it to break. 

On the model, I need to know what you want to model.
